flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerFactory.java

```diff
@@ -47,10 +48,12 @@
 			FatalErrorHandler fatalErrorHandler,
 			ClusterInformation clusterInformation,
 			@Nullable String webInterfaceUrl,
-			ResourceManagerMetricGroup resourceManagerMetricGroup) throws Exception {
+			MetricRegistry metricRegistry,
+			String hostname) throws Exception {
 
 		final ResourceManagerRuntimeServices resourceManagerRuntimeServices = createResourceManagerRuntimeServices(
-			configuration, rpcService, highAvailabilityServices);
+			configuration, rpcService, highAvailabilityServices, metricRegistry, hostname);
```

Comment:
I think we could pass in the `SlotManagerMetricGroup` at this point. This would have the benefit that we would not have to forward the `metricRegistry` and the `hostname` all the way.